homeassistant/components/nobo_hub/config_flow.py

```diff
@@ -0,0 +1,252 @@
+    def _device(hub):
+        (ip, serial_prefix) = hub
+        return {"ip": ip, "serial_prefix": serial_prefix}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if self._discovered_hubs is None:
+            self._discovered_hubs = list(
+                map(self._device, await nobo.async_discover_hubs(loop=self.hass.loop))
+            )
+
+        if len(self._discovered_hubs) == 0:
+            # No hubs auto discovered
+            return await self.async_step_manual()
+
+        errors = {}
+        serial_suffix = ""
+        if user_input is not None:
+            if "manual" in user_input:
+                return await self.async_step_manual()
+            serial, ip_address = None, None
+            if "serial_suffix" not in user_input or user_input["serial_suffix"] == "":
+                errors["base"] = "missing_serial_suffix"
+            else:
+                hub = self._discovered_hubs[user_input["device"]]
+                serial_prefix = hub["serial_prefix"]
+                serial_suffix = user_input["serial_suffix"]
+                serial = f"{serial_prefix}{serial_suffix}"
+                if "store_ip" in user_input and user_input["store_ip"]:
+                    ip_address = hub["ip"]
+
+            if not errors:
+                try:
+                    return await self._create_configuration(serial, ip_address)
+                except NoboHubConnectError as error:
+                    errors["base"] = error.msg
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional("manual"): bool,
+                vol.Required("device"): vol.In(self._hubs()),
+                vol.Optional("serial_suffix", default=serial_suffix): str,
+                vol.Optional("store_ip"): bool,
+            }
+        )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors,
+        )
+
+    async def async_step_manual(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle manual configuration. Triggered if no devices are discovered or by user."""
+        errors = {}
+        serial, ip_address = None, None
+        if user_input is not None:
+            serial = user_input.get(CONF_SERIAL)
+            ip_address = user_input.get(CONF_IP_ADDRESS)
+            try:
+                return await self._create_configuration(serial, ip_address)
+            except NoboHubConnectError as error:
+                errors["base"] = error.msg
+
+        return self.async_show_form(
+            step_id="manual",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_SERIAL, default=serial): str,
+                    vol.Required(CONF_IP_ADDRESS, default=ip_address): str,
+                }
+            ),
+            errors=errors,
+        )
```

Comment:
```suggestion
        user_input = user_input or {}
        return self.async_show_form(
            step_id="manual",
            data_schema=vol.Schema(
                {
                    vol.Required(CONF_SERIAL, default=user_input.get(CONF_SERIAL)): str,
                    vol.Required(
                        CONF_IP_ADDRESS, default=user_input.get(CONF_IP_ADDRESS)
                    ): str,
                }
            ),
            errors=errors,
        )
```

Reply:
Why use `user_input.get(CONF_SERIAL)` here when you suggested to use `user_input[CONF_SERIAL]` above?